geode-core/src/main/java/org/apache/geode/internal/cache/backup/BackupManager.java

```diff
@@ -15,476 +15,85 @@
 package org.apache.geode.internal.cache.backup;
 
 import java.io.File;
-import java.io.IOException;
-import java.util.Collection;
-import java.util.HashMap;
 import java.util.HashSet;
-import java.util.LinkedList;
 import java.util.List;
-import java.util.Map;
 import java.util.Set;
-import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
 
-import org.apache.commons.io.FileUtils;
-import org.apache.logging.log4j.Logger;
-
-import org.apache.geode.InternalGemFireError;
-import org.apache.geode.cache.DiskStore;
 import org.apache.geode.cache.persistence.PersistentID;
 import org.apache.geode.distributed.internal.DistributionManager;
 import org.apache.geode.distributed.internal.MembershipListener;
 import org.apache.geode.distributed.internal.membership.InternalDistributedMember;
-import org.apache.geode.internal.cache.DirectoryHolder;
 import org.apache.geode.internal.cache.DiskStoreBackup;
 import org.apache.geode.internal.cache.DiskStoreImpl;
-import org.apache.geode.internal.cache.GemFireCacheImpl;
 import org.apache.geode.internal.cache.InternalCache;
-import org.apache.geode.internal.cache.Oplog;
-import org.apache.geode.internal.logging.LogService;
 
-/**
- * This class manages the state an logic to backup a single cache.
- */
 public class BackupManager {
-  private static final Logger logger = LogService.getLogger();
 
-  static final String INCOMPLETE_BACKUP_FILE = "INCOMPLETE_BACKUP_FILE";
-
-  private static final String BACKUP_DIR_PREFIX = "dir";
-  private static final String DATA_STORES_DIRECTORY = "diskstores";
   public static final String DATA_STORES_TEMPORARY_DIRECTORY = "backupTemp_";
-  private static final String USER_FILES = "user";
-
+  private final ExecutorService executor = Executors.newSingleThreadExecutor();
```

Comment:
Use threadFactory to have daemon thread?

Reply:
added